editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search/Headline.tsx

```diff
@@ -15,17 +15,21 @@ export const Headline: React.FC<Props> = ({ text, matchIndices, variant }) => {
   return (
     <>
       {text.split("").map((char, index) => (
-        <Typography
-          component="span"
-          variant={variant}
-          key={`headline-character-${index}`}
-          sx={(theme) => ({
-            fontWeight: isHighlighted(index) ? FONT_WEIGHT_BOLD : "regular",
-            fontSize: theme.typography.body2.fontSize,
-          })}
-        >
-          {char}
-        </Typography>
+        <>
+          <Typography
+            component="span"
+            variant={variant}
+            key={`headline-character-${index}`}
+            sx={(theme) => ({
+              fontWeight: isHighlighted(index) ? FONT_WEIGHT_BOLD : "regular",
+              fontSize: theme.typography.body2.fontSize,
+            })}
+          >
+            {char}
+          </Typography>
+          {/* Add wordbreak after special characters */}
+          {char.match(/\W/) && <wbr />}
```

Comment:
I don't fully understand why this is _outside_ of the `<Typography>` block and only within the fragment? Don't we want all characters to share the styles above (eg indexed/matching characters _after_ a special character should still have bold highlighting?)

Reply:
Good question! This is a little unexpected as the headline breaks each word into characters and inserts a "word" break after them. The `<wbr/>` is just a non-visual element so that it splits in the right place. 

The reason there's only one word highlighted is that the search is configured to just return the first match from a string, not all.

Reply:
<img width="322" alt="image" src="https://github.com/user-attachments/assets/bba383ef-c2b8-44b6-a5f3-b4c6bc762074">